README.md

```diff
@@ -0,0 +1,77 @@
+# chavalrous-tortoise
+
+  team-size: 1 - Abraham Ferguson  
+
+#Advanced Data Structures
+
+#Hash Table
+
+  Maps keys to values, like a dictionary or a phone book. Or an object in JavaScript...
+
+  A data structure used to implement an associative array, a structure that can map keys to values. A hash table uses a hash function to compute an index into an array of buckets or slots, from which the desired value can be found.
+    []const ht = new HashTable()
+    []ht.put("name", "Zanzibar")  // adds a key-value pair to the hash table.
+    []ht.get("name")              // returns the data associated with key.
+    []ht.contains("name")         // returns true if the hash table contains the key.
+    []ht.iterate((k, v) => console.log(`${k}: ${v}`)) // takes a callback function and passes it each key and value in sequence.
+    []ht.remove("name")           // removes a key-value pair by key.
+    []ht.size()                   // returns the number of key-value pairs in the hash table.
+    []HashTable.hash("name")      // generates a hash for the key "name"
+
```

Comment:
The formatting on these checklists is a bit wonky.  They don't display right when parsed.  Unchecked boxes should contain a space, and I think we generally make them a list item.

Also, don't forget to check off the checkboxes for things you completed!